src/components/Statuses/Statuses.tsx

```diff
@@ -23,10 +25,45 @@ const Statuses = () => {
     return <div className={styles.base}>Keine Daten vorhanden</div>;
   }
 
+  //sort by date
+  const sortedStatuses: { dateKey: string; statuses: Status[] }[] = [];
+  statuses.forEach((status) => {
+    const departureSchedule = parseSchedule({
+      actual:
+        status.train.origin.departureReal ??
+        status.train.origin.departure ??
+        '',
+      planned: status.train.origin.departurePlanned ?? '',
+    });
+    const date = new Date(departureSchedule.actualValue);
+    const dateKey = `${date.getDate()}.${
+      date.getMonth() + 1
+    }.${date.getFullYear()}`;
+
+    const existingDate = sortedStatuses.find((s) => s.dateKey === dateKey);
+    if (existingDate) {
+      existingDate.statuses.push(status);
+    } else {
+      sortedStatuses.push({
+        dateKey,
+        statuses: [status],
+      });
+    }
+  });
```

Comment:
Technically you are more _grouping_ here than _sorting_. So personally I would rather create something named `groupedStatuses` and make it a map of the date to the array of statuses from each day.

I like that you've used the schedule utility, but there is not much going on here so you could directly get the grouping key from `new Date(status.train.origin.departureReal ?? status.train.origin.departure ?? '')`. Preferably this would happen in a helper function defined outside of the loop.